### ISSUE TITLE
NoSuchMethodError: The getter 'keys' was called on null.

### ISSUE THREAD

**Issue by JoakimMellonn** [aws-amplify/amplify-flutter#3590]
### Description

I'm getting the above error while identifying a user with Push Notifications. The user is still identified in Pinpoint, so this doesn't actually create any problems for me, but it's still showing up every time. I have tested it both on Android and iOS, both on physical devices.

Here is the full error log:
```zsh
I/flutter (13970): An error occurred while requesting notification permissions: NoSuchMethodError: The getter 'keys' was called on null.
I/flutter (13970): Receiver: null
I/flutter (13970): Tried calling: keys
```

Here is how I'm identifying the user:
```dart
final user = await Amplify.Auth.getCurrentUser();
final userProfile = AWSPinpointUserProfile(
  name: context.read<AuthAppProvider>().fullName,
  email: context.read<AuthAppProvider>().email,
);

await Amplify.Notifications.Push.identifyUser(
  userId: user.userId,
  userProfile: userProfile,
);
```

### Categories

- [ ] Analytics
- [ ] API (REST)
- [ ] API (GraphQL)
- [ ] Auth
- [ ] Authenticator
- [ ] DataStore
- [X] Notifications (Push)
- [ ] Storage

### Steps to Reproduce

1. Create Flutter app with Push notifications enabled.
2. Identify user, the same way as shown in the docs.

### Platforms

- [X] iOS
- [X] Android
- [ ] Web
- [ ] macOS
- [ ] Windows
- [ ] Linux

### Flutter Version

3.10.6

### Amplify Flutter Version

1.3.3

### Deployment Method

Amplify CLI

**Comment by jamilsaadeh97** [comment]
@JoakimMellonn  I had the same issue and I realized that the AWSPinpointUserProfile should have the userAttributes passed in the parameters (in my case I passed an empty map).

**Comment by JoakimMellonn** [comment]
> @JoakimMellonn I had the same issue and I realized that the AWSPinpointUserProfile should have the userAttributes passed in the parameters (in my case I passed an empty map).

That did also work for me, but it's still a kinda weird error message.

**Comment by Equartey** [comment]
@JoakimMellonn sorry you ran into this error. 

This is a bug and I have opened a PR to address it. 

In the mean time, y'all have identified the work around -- pass a value for `userAttributes` even if its an empty map `{}`. Alternatively, consider using the type `UserProfile` in place of `AWSPinpointUserProfile` where applicable.

**Comment by Equartey** [comment]
Hey @JoakimMellonn, a fix for this was included in our latest release [1.4.0](https://github.com/aws-amplify/amplify-flutter/releases/tag/v1.4.0). Thanks again for identifying it!
